Import Libraries

In [1]:
import pickle
import json
import pandas as pd
import numpy as np


Memanggil Model yg sudah di save

In [2]:
with open('model_lin_reg.pkl', 'rb') as file_1:
  model_lin_reg = pickle.load(file_1)


with open('model_encoder.pkl', 'rb') as file_2:
  model_encoder = pickle.load(file_2)

with open('list_num_cols.txt', 'r') as file_3:
  list_num_cols = json.load(file_3)

with open('list_cat_cols.txt', 'r') as file_4:
  list_cat_cols = json.load(file_4)

Membuat dan memasukan entry baru untuk di test 

In [4]:
data_inf = {
    'id  ': '5b21cd69-43fb-4a1d-a42e-8b3eb54c0634566jg', 
    'timestamp': '1543732386.987', 
    'hour   ': 3, 
    'day' : 17,
    'month': 11,
    'datetime ': '2018-12-01 08:1:09', 
    'timezone  ': 'America/New_York', 
    'source':  'North End', 
    'destination  ' : ' North Station',
    'surge_multiplier': 1,
    'cab_type': 'Lyft', 
    'name': 'Lyft XL', 
    'distance': 3, 
    'latitude ' : 42.3505,
    'longitude': -71.1054,
    'temperature':  37.96, 
    'apparentTemperature':  37.96, 
    'short_summary ': ' Overcast ', 
    'long_summary ' : 'Light rain until evening',
    'precipIntensity': 0,
    'precipProbability  ': 0, 
    'humidity ': 0.9, 
    'windSpeed  ': 3, 
    'windGust' : 4.5,
    'windGustTime  ': 1543755700,
    'visibility': 7.6, 
    'temperatureHigh':45 , 
    'temperatureHighTime ': 1543784400, 
    'temperatureLow' : 44.79,
    'temperatureLowTime': 1543816800,
    'icon ': 'cloudy', 
    'dewPoint ': 38, 
    'pressure ': 1011.23, 
    'windBearing  ' :  135,
    'cloudCover ': 1.9,
    'uvIndex  ': 0, 
    'visibility.1':  9.706, 
    'ozone': 269.6, 
    'sunriseTime' :  1543751804,
    'sunsetTime ':  1543785248,
    'moonPhase': 0.86, 
    'precipIntensityMax': 1543770000, 
    'uvIndexTime ': 3, 
    'temperatureMin ' :  36.1,
    'temperatureMinTime  ':  1543726800,
    'temperatureMax  ': 50.37, 
    'temperatureMaxTime ': 1543784400, 
    'apparentTemperatureMin': 38.88, 
    'apparentTemperatureMinTime   ' : 1543748400,
    'apparentTemperatureMax': 50,
    'apparentTemperatureMaxTime ': 1543784400,
    
}
data_inf = pd.DataFrame([data_inf])
data_inf 

id         timestamp  hour     day  \
0  5b21cd69-43fb-4a1d-a42e-8b3eb54c0634566jg  1543732386.987        3   17   

   month           datetime         timezone       source   destination    \
0     11  2018-12-01 08:1:09  America/New_York  North End   North Station   

   surge_multiplier  ... precipIntensityMax uvIndexTime   temperatureMin   \
0                 1  ...         1543770000            3             36.1   

   temperatureMinTime    temperatureMax    temperatureMaxTime   \
0            1543726800             50.37           1543784400   

   apparentTemperatureMin apparentTemperatureMinTime     \
0                   38.88                    1543748400   

  apparentTemperatureMax  apparentTemperatureMaxTime   
0                     50                   1543784400  

[1 rows x 51 columns]

Split Data

In [5]:
data_inf_num = data_inf[list_num_cols]
data_inf_cat = data_inf[list_cat_cols]
data_inf_num

distance  surge_multiplier
0         3                 1

Scaling dan Encoding data kemudian di gabungkan 

In [6]:
data_inf_cat_encoded = model_encoder.transform(data_inf_cat)
data_inf_final = np.concatenate([data_inf_num, data_inf_cat_encoded], axis=1)

Memasukan data kepada model yang ada , untuk di lihat prediksi nya

In [7]:
y_pred_inf = model_lin_reg.predict(data_inf_final)
y_pred_inf


array([16.92614746])

Jadi prediksi harga input yang di atas adalah $16.9 